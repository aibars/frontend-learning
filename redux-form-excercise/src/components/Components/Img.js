import React from 'react';
class Img extends React.Component {

    render() {
        return (
            <img src={this.props.values.src} alt={this.props.values.alt}></img>
        );
    }
}

export default Img;