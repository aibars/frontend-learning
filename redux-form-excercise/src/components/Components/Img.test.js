import React from 'react';
import { shallow } from 'enzyme';

import Img from './Img';

describe('Img', () => {
  it('works ok', () => {
    const component = shallow((
      <Img
        values={{
          alt: 'Cool image',
          src: 'http://lorempixel.com/400/200',
        }}
      />
    ));

    expect(component).toMatchSnapshot();
  });
});
