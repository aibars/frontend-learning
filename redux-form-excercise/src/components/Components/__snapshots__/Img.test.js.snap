// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Img works ok 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Img
    values={
      Object {
        "alt": "Cool image",
        "src": "http://lorempixel.com/400/200",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "alt": "Cool image",
      "src": "http://lorempixel.com/400/200",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "Cool image",
        "src": "http://lorempixel.com/400/200",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
